day-01/y_part-1.cpp

```diff
@@ -0,0 +1,24 @@
+#include <algorithm>
+#include <iostream>
+#include <vector>
+
+int main() {
+
+  std::vector<int> left;
+  std::vector<int> right;
+
+  int l, r;
+  while (std::cin >> l >> r) {
+    left.push_back(l);
+    right.push_back(r);
+  }
+
+  std::sort(left.begin(), left.end());
+  std::sort(right.begin(), right.end());
+  int sum = 0;
+  // why do I need size_t here?
+  for (std::size_t i = 0; i < left.size(); i++)
```

Comment:
>  // why do I need size_t here?

`.size()` returns an unsigned integer length type (i.e. size_t). If you were to use `int` for your counter variable, a conversion would need to happen. This is risky as the target type cannot represent all possible input values for two reasons:
1. If you're on a 64-bit system, a `size_t` is usually 64 bits, but `int` is only 32. 
1. Even if they were the same number of bits, the signed-ness takes away one bit. This shifts half of the possible values to the negative. 

These conversions are error-prone and the results are not relyable across different compilers. That's why there's a best-practice to let the compiler warn if it has to do such a conversion (in gcc/clang enabled with the compiler flag `-Wconversion`).

There are four solutions to this problem: 
1. Turn off the warning and live with the risk. Maybe you can be sure that the size will never be big enough for this to matter.
1. Use the same (unsigned) type for your loop counter as `.size()` returns.
1. Use `std::ssize()` that returns a signed size type. 
1. Replace the manual loop with an algorithm. This removed the need for a manual counter variable altogether.